Client.js

```diff
@@ -4,7 +4,7 @@ var d = require('describe-property');
 var EventEmitter = require('events').EventEmitter;
 var appendHashToArray = require('./utils/appendHashToArray');
 var parseInfo = require('./utils/parseInfo');
-var Promise = require('./utils/Promise');
+if (typeof Promise === 'undefined') var Promise = Promise || require('bluebird');
```

Comment:
Hello there,
not a maintainer here whatsoever but this won't work.

You're saying that if Promise is undefined, then you want Promise to be undefined, otherwise require('bluebird');
You'll end up always requiring bluebird, whatever happens here.

Also, this had probably been better kept in the utils/Promise.js since would have allowed you not to repeat yourself everytime you need a Promise.